src/vcpkg/base/messages.cpp

```diff
@@ -250,6 +250,28 @@ namespace vcpkg::msg
 
     ::size_t detail::number_of_messages() { return messages().names.size(); }
 
+    std::string detail::format_examples_for_args(StringView extra_comment,
```

Comment:
This whole thing is just to fix trailing whitespace in the comments of localization messages. It was sucked out into a .cpp (after being flattened from template args to a struct in the header) so that it could use `Strings::join` without creating header circularity or making `messages.h` (which is included in everything now) huge.